code/API_definitions/connectivity-insights-subscriptions.yaml

```
@@ -0,0 +1,1265 @@
+        packetDelayBudget:
+          $ref: "#/components/schemas/PolicyFulfillmentConfidence"
+        targetMinDownstreamRate:
+          $ref: "#/components/schemas/PolicyFulfillmentConfidence"
+        targetMinUpstreamRate:
+          $ref: "#/components/schemas/PolicyFulfillmentConfidence"
+        packetlossErrorRate:
+          $ref: "#/components/schemas/PolicyFulfillmentConfidence"
+        jitter:
+          $ref: "#/components/schemas/PolicyFulfillmentConfidence"
+        additionalKpis:
+          $ref: "#/components/schemas/AdditionalKpis"
+
+    PolicyFulfillmentConfidence:
+      type: string
+      description: |
+        a plain-language indicator of how confident the network is to meet a
+        given network demand.
+      enum:
+        - meets the application requirements
+        - unable to meet the application requirements
+
+    ApplicationProfileId:
+      description: Identifier for the Application Profile
+      type: string
+      format: uuid
+
+    Source:
+      type: string
+      format: uri-reference
+      minLength: 1
+      description: |
+        Identifies the context in which an event happened, as a non-empty
+        `URI-reference` like:
+        - URI with a DNS authority:
+          * https://github.com/cloudevents
+          * mailto:cncf-wg-serverless@lists.cncf.io
+        - Universally-unique URN with a UUID:
+          * urn:uuid:6e8bc430-9c3a-11d9-9669-0800200c9a66
+        - Application-specific identifier:
+          * /cloudevents/spec/pull/123
+          * 1-555-123-4567
+      example: "https://notificationSendServer12.supertelco.com"
+    CloudEvent:
+      description: The Cloud-Event used for the callback.
+      required:
+        - id
+        - source
+        - specversion
+        - type
+        - time
+      properties:
+        id:
+          type: string
+          description: |
+            identifier of this event, that must be unique in the source context.
+        source:
+          $ref: "#/components/schemas/Source"
+        type:
+          $ref: "#/components/schemas/Event-typeNotification"
+        specversion:
+          type: string
+          description: |
+            Version of the specification to which this event conforms (must be
+             1.0 if it conforms to cloudevents 1.0.2 version)
+          enum:
+            - "1.0"
+        datacontenttype:
+          type: string
+          description: |
+           'media-type that describes the event payload encoding,
+           must be "application/json" for CAMARA APIs'
+          enum:
+            - application/json
+        data:
+          $ref: "#/components/schemas/NetworkQualityInsight"
+        time:
+          $ref: "#/components/schemas/DateTime"
+      discriminator:
+        propertyName: "type"
+        mapping:
+          org.camaraproject.connectivityinsights.v0.networkQuality-event:
+            "#/components/schemas/NetworkQualityTypeEvent"
```

It looks that each event type has separate component defined:
```suggestion
          org.camaraproject.connectivity-insights-subscriptions.v0.network-quality-met:
            "#/components/schemas/EventNetworkQualityMet"
           org.camaraproject.connectivity-insights-subscriptions.v0.network-quality-not-met:
            "#/components/schemas/EventNetworkQualityNotMet"
```
Following suggestions for lines 891-898

---

@rartych please review this accept of the change in the latest commit. i have reduced it back to 2 events only. first to notify on the network quality and other for subscription end. 
In the same EventNetworkQuality the component support enum of either meets the requirement or does not meet the requirement for each network KPI defined. So 2 different components are not required. 

I will create a separate issue to track this, rather than marking this unresolved. 
issue #95 created for tracking this.